docs/core/deploying/preparing-libraries-for-trimming.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Preparing libraries for trimming
+description: Learn how to prepare libraries for trimming.
+author: sbomer
+ms.author: svbomer
+ms.date: 08/25/2020
+---
+
+# Preparing libraries for trimming
+
+## Trim warnings in apps
+
+When publishing an app, `PublishTrimmed` will produce trim analysis [warnings](trimming-options.md#analysis-warnings) (in .NET 6+) for patterns that are not statically understood to be compatible with trimming, including patterns in your code and in dependencies.
+
+You will encounter detailed warnings originating from your own code and `ProjectReference` dependencies. You may also see warnings like `warning IL2104: Assembly 'SomeAssembly' produced trim warnings` for `PackageReference` libraries. This means that the library contained patterns which are not guaranteed to work in the context of the trimmed app, and may result in a broken app. Consider contacting the author to see if the library can be annotated for trimming.
+
+To resolve warnings originating from the app code, jump ahead to the instructions on [resolving trim warnings](#resolving-trim-warnings). If you are interested in making your own `ProjectReference` libraries trim friendly, follow the instructions to [enable library trim warnings](#enabling-library-trim-warnings).
+
+If your app only uses parts of a library that are compatible with trimming, consider [enabling trimming](trimming-options.md#trim-additional-assemblies) of this library if it is not already being trimmed. This will only produce warnings if your app uses problematic parts of the library. (You can also [show detailed warnings](trimming-options.md#showing-detailed-warnings) for the library to see which parts of it are problematic.)
+
+## Enabling library trim warnings
+
+These instructions show how to enable and resolve static analysis warnings to prepare a library for trimming. Follow these steps if you are authoring a library and either want to proactively make your library trimmable, or have been contacted by app authors who encountered trim warnings from your library.
+
+Use the .NET 6 SDK for the best experience. It is possible to [show analysis warnings](trimming-options.md#analysis-warnings) in .NET 5, but not all framework assemblies were annotated correctly in .NET 5. Also, this will show detailed warnings for every library, including framework libraries. These instructions assume you are using the .NET 6 SDK.
+
+### Incomplete Roslyn analyzer
+
+During development, you may set `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` (in .NET 6+) in your library project to get a _limited_ set of warnings from the Roslyn analyzer. This analyzer is incomplete and should only be used as a convenience. It is important to follow the next steps to ensure that your library is compatible with trimming.
```

Comment:
This makes us look pretty bad - to soften this a bit I would add that we're working on improving the analyzer.

Reply:
In fact I would probably change this section. (Rewrite in real english please :wink:)

### Enable analysis across toolchain

You should set `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` in your library project. This will not have any effect on the output, but it will enable trim analysis across the entire toolchain.

It will enable the Roslyn analyzer for trimming, which is currently not completely, but it's useful for the cases where it works. It provides immediate response from within IDE or build. It doesn't not report all trim analysis warnings and doesn't cover all patterns though. The analyzer should improve over time to provide wider coverage of the trim incompatible patterns.

Reply:
Reworded, hopefully this is close to what you had in mind.